### PR TITLE
Add first draft of mentoring notes for Two Fer exercise in Go

### PR DESCRIPTION
**Pull request by kytrinyx**
These are more observations than hard-and-fast rules.

I'd love to hear other Go mentors' thoughts on this.

- What do you disagree with?
- What resources do you like pointing people to?
- What have you observed that I'm missing?

View the rendered notes to read comfortably: https://github.com/exercism/mentors/blob/go-two-fer-mentoring-notes/tracks/go/exercises/two-fer/README.md
If you have line-by-line comments, the files tab is great for that: https://github.com/exercism/mentors/pull/4/files

### CODE REVIEW

**Comment by junedev** [comment]
Resembles my thoughts. I didn't push students towards `fmt` so far but I agree it would make sense.
Thanks for the nice write up!

**Comment by F3PiX** [comment]
I am not a Go mentor, so I read this not from a Go mentor point of view, but from the point of view of repeatability for other exercises and tracks. 
This looks really helpful!! I love it. 😍 

And if it stays like this, I still love it. Just some thoughts that came to mind thinking about this: 

The introduction, where this exercise comes in the range, is already helpful. (I wouldn't mind if there was also a general description of the goal of this exercise. Even if it is just what you stated: first acquaintance.)    

I see that you are not recommending a MVS™ ( minimal viable solution ;-) ) for approval. And I'd suspect that is on purpose, right? 
However, did you consider to be a _bit_ more instructive for mentors who are starting mentoring or are uncertain about it? I am asking, because I'd think that they would be happy with a bit more guidance. I don't mean setting a hard rule, but just giving some rules of thumb, like:
"When in doubt, address the formatting plus one of the above mentioned points". I am sure you could formulate that much better than I can :-)

I really like the sprinkles of mentoring wisdom like "Gently remind them that this adds mental overhead for the reader, since we have to read all of it before deciding that it's irrelevant." I think this is something that really should be transferred to other exercise documents. So, maybe it helps to give it a bit more emphasis, like:
*Mentoring Tip:* ...sprinkle of wisdom... 

For further reference it may come in handy to add numbering to the points you mention, and maybe collect them under a subtitle "Focus Points" or something. 

One of the things I am thinking about in my own mentoring comments, is addressing each of three things:
1) Readability (your points: 'comments' and 'formatting')
2) Some language specific perks, like syntax or ...    ( I guess your points 'String concatenation" and the compound parameter name)
3) Language specific or even general coding habits ( I guess what you mention under "The Conditional)

I tend to _ask_ to solve the things under 1), _give_ them a little present for 2) and _challenge_ them for 3).  Not saying that that is the way to go, but would you think the general distinction can be helpful in a 
document like this? I am not sure, but I feel there is something interesting hidden in the distinction :joy:

Thanks!!
Maud

File: tracks/go/exercises/two-fer/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+One mentor recommended to focus on a single thing, and only add other improvements as "food for thought" that the author can take with them as they move on to the next exercise.
+
+**The Conditional**
+* Is there duplicated content? Sometimes this will be two branches, sometimes this will be one branch with the whole phrase in it, and then the phrase repeated after the conditional. Here you can suggest getting the smallest possible difference isolated into the conditional.
+* Are they re-assigning the param to a local variable? You can point out that they have two variables doing the same thing, e.g. `input` and `value` are both doing the same thing. Can you get away with just having one of them?
+
+**String Concatenation**
+* are they using a slice and strings.Join()? Point them towards `fmt` package as the more conventional choice here. Maybe not point straight to which function, it's often useful to let people explore a bit and find it on their own.
+* are they using `+` to concatenate? Same thing, but a more gentle pointer since it's perfectly legit to see `+`, just more common to see `fmt`, and worth getting used to.
```
Comment (hilary) [inline tracks/go/exercises/two-fer/README.md]: Depending on how it is written, using `+` is also far, far faster than `fmt`. I tend to reserve `fmt` for times I need it, as it is comparatively expensive.

**Comment by gabeguz** [comment]
Just a general comment on using `+` in this particular instance... it keeps the code a bit cleaner, I think since you don't need any imports, just a simple `return "one for " + name " one for me"`. In any case, that's why I haven't been pushing the use of `fmt` on this exercise.

**Comment by kytrinyx** [comment]
I made some tweaks based on your feedback, and am merging it in... would *love* PRs to improve it!

**Pull request merged**